phalcon/di/service/serviceregistry.zep

```diff
@@ -0,0 +1,115 @@
+namespace Phalcon\Di\Service;
+
+use Phalcon\DiInterface;
+use Phalcon\Di\Exception;
+use Phalcon\Di\ServiceInterface;
+use Phalcon\Config\Adapter\Yaml;
+
+/**
+ * Phalcon\Di\Service\ServiceRegistry
+ *
+ **/
+class ServiceRegistry
+{
+
+	/**
+	 *
+	 */
+	protected di;
+
+	/**
+	 *
+	 */
+	public function __construct(<DiInterface> di)
+	{
+		let this->di = di;
+	}
+
+	/**
+	 *
+	 */
+	public function registerDir(dirPath)
+	{
+		var dirPathItem;
+
+		switch (gettype(dirPath)) {
+			case "string":
+				this->_registerDir(dirPath);
+				break;
+			case "array":
+			case "object":
+				for dirPathItem in iterator(dirPath) {
+					this->_registerDir(dirPathItem);
+				}
+				break;
+			default:
+				throw new Exception("Invalid services.");
+				break;
+		}
+	}
+
+	/**
+	 *
+	 */
+	protected function _registerDir(string! dirPath)
```

Comment:
Method names should not be prefixed with a single underscore to indicate protected nature. Actually when you extend this class in PHP your code will not pass the code style tests.

A propose get rid of underscores in method names for 4.x

Reply:
I did it that way since I wanted the method name to be the exact same.  What do you think about `registerDirInternal`?

Reply:
@dschissler Frankly speaking I'm a little confused with `Dir`. Probably it's because I'm not a native speaker (like many of our users). Dir it is: DIRection. DIRectory, Do It Right, ...? Why do we need abbreviations for methods that aren't part of the public API?

Reply:
Yes `Dir` is short for `Do it right`.  Its known as IDD (Intent Driven Development).  Saying nice things and that sort of thing.

Kevin and I were chatting about it and I'll be doing more API work in about 7 days at which time I'll take down this pull request.